pyfr/h5writer.py

```diff
@@ -10,46 +10,48 @@
 from pyfr.mpiutil import get_comm_rank_root
 
 
-class H5Writer(BaseIntegrator):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        # Base output directory and file name
-        self._basedir = self.cfg.getpath('soln-output', 'basedir', '.')
-        self._basename = self.cfg.get('soln-output', 'basename', raw=True)
+class H5Writer(object):
+    def __init__(self, intg, basedir, basename, prefix, serial_h5=False):
+        # Base output directory and file name and prefix for the data-set.
+        self._basedir = basedir
+        self._basename = basename
+        self._prefix = prefix
 
         # Output counter (incremented each time output() is called)
         self.nout = 0
 
+        # The float type and the mesh uuid
+        self.fpdtype = intg.backend.fpdtype
+
         # MPI info
         comm, rank, root = get_comm_rank_root()
 
         # Get the type and shape of each element in the partition
-        etypes, shapes = self.system.ele_types, self.system.ele_shapes
+        etypes, shapes = intg.system.ele_types, intg.system.ele_shapes
 
-        # Gather this information onto the root rank
-        eleinfo = comm.gather(zip(etypes, shapes), root=root)
+        # Gather this information and distribute
+        eleinfo = comm.allgather(zip(etypes, shapes))
 
         # Deciding if parallel
         parallel = (h5py.get_config().mpi and
                     h5py.version.version_tuple >= (2, 5) and
-                    not self.cfg.getbool('soln-output', 'serial-h5', False))
+                    not serial_h5)
 
         if parallel:
             self._write = self._write_parallel
+            self._loc_names = loc_names = []
+            self.all_datalist = []
```

Comment:
Not so sure about this name.  Also, the member is public whereas most of the other variables appear to be 'private'.